docs/design/SYSTEM_DESIGN.md

```diff
@@ -6,101 +6,138 @@
+GatewayClass controllerName, configure a Provider, etc. Currently, Envoy Gateway only supports runtime configuration
+through a configuration file. See the [configuration](../CONFIG.md) guide for additional details.
+
+#### Dynamic Configuration
+Dynamic configuration is based on the concept of a declaring the desired state of the data plane and using
+reconciliation loops to drive the actual state toward the desired state. The desired state of the data plane is
+defined as Kubernetes resources that provide the following services:
+* Infrastructure Management- Manage the data plane infrastructure, i.e. deploy, upgrade, etc. This configuration is
+  expressed through [GatewayClass][gc] and [Gateway][gw] resources. A TBD [Custom Resource][cr] can be referenced by
+  `gatewayclass.spec.parametersRef` to modify data plane infrastructure default parameters,
+  e.g. expose Envoy network endpoints using a NodePort service instead of a LoadBalancer service.
+* Traffic Routing- Define how to handle application-level requests to backend services. For example, route all HTTP
+  requests for "www.example.com" to a backend service running a web server. This configuration is expressed through
+  [HTTPRoute][hroute] and [TLSRoute][troute] resources that match, filter, and route traffic to a [backend][be].
+  Although a backend can be any valid Kubernetes Group/Kind resource, Envoy Gateway only supports a [Service][svc]
+  reference.
 
 ### Components
 
-#### Config Sources
-This component is responsible for consuming the user configuration from various platforms. Data persistence should be tied to the specific config source’s capabilities. For e.g. in Kubernetes, the resources will persist in `etcd`, if using the `path-watcher`, the resources will persist in a file.
-
-##### Kubernetes
-It watches the Kubernetes API Server for resources, fetches them, and publishes it to the translators for further processing.
-
-##### Path Watcher
-It watches for file changes in a path, allowing the user to configure Envoy Gateway using resource configurations saved in a file or directory.
-
-##### Config Server
-This is a HTTP/gRPC Server allowing Envoy Gateway to be configured from a remote endpoint. 
+#### Provider
+A Provider is an infrastructure component that Envoy Gateway calls to establish its
+[static configuration](#static-configuration), resolve services, persist data, etc. Kubernetes and File are the only
+supported providers. However, other providers can be added in the future as Envoy Gateway use cases are better
+understood. A provider is configured at runtime through Envoy Gateway's [static configuration](#static-configuration).
+
+##### Kubernetes Provider
+* Uses Kubernetes-style controllers to reconcile Kubernetes resources that comprise the
+  [dynamic configuration](#dynamic-configuration).
+* Manages the data plane through Kubernetes API CRUD operations.
+* Uses Kubernetes for Service discovery.
```

Comment:
Where did we settle on the discussion of having service discovery be a Provider configuration? I'm just thinking about the Consul use-case where we might provision Kubernetes infrastructure but use Consul for service discovery... or Consul with a File Provider.

Reply:
From my understanding, we settled on a Provider providing:
- Proxy infra management.
- Service discovery.
- Data persistence (if needed)
- other future services (if needed).

We can make these Provider sub-components configurable in the future (if needed). For example, the Kubernetes provider will use standard kube-dns service discovery semantics but specifying `controlplane.provider.kubernetes.svcDisc.type.Consul` in the static config will cause EG to use Consul for service discovery.

Reply:
I'm in the process of updating https://github.com/envoyproxy/gateway/pull/95 to address this comment in more detail.